src/titiler/xarray/titiler/xarray/factory.py

```diff
@@ -0,0 +1,364 @@
+                ),
+            ],
+            tileMatrixSetId: Annotated[
+                Literal[tuple(self.supported_tms.list())],
+                Path(
+                    description="Identifier selecting one of the TileMatrixSetId supported."
+                ),
+            ],
+            scale: Annotated[
+                int,
+                Field(
+                    gt=0, le=4, description="Tile size scale. 1=256x256, 2=512x512..."
+                ),
+            ] = 1,
+            format: Annotated[
+                ImageType,
+                "Default will be automatically defined if the output image needs a mask (png) or not (jpeg).",
+            ] = None,
+            multiscale: Annotated[
+                Optional[bool],
+                Query(
+                    title="multiscale",
+                    description="Whether the dataset has multiscale groups (Zoom levels)",
+                ),
+            ] = None,
+            src_path=Depends(self.path_dependency),
+            reader_params=Depends(self.reader_dependency),
+            tile_params=Depends(self.tile_dependency),
+            layer_params=Depends(self.layer_dependency),
+            dataset_params=Depends(self.dataset_dependency),
+            post_process=Depends(self.process_dependency),
+            rescale=Depends(self.rescale_dependency),
+            color_formula=Depends(self.color_formula_dependency),
+            colormap=Depends(self.colormap_dependency),
+            render_params=Depends(self.render_dependency),
+            env=Depends(self.environment_dependency),
+        ):
+            """Create map tile from a dataset."""
+            tms = self.supported_tms.get(tileMatrixSetId)
+
+            reader_options = reader_params.as_dict()
+            if getattr(tile_params, "multiscale", False):
+                reader_options["group"] = z
+
+            with rasterio.Env(**env):
+                with self.reader(src_path, tms=tms, **reader_options) as src_dst:
+                    image = src_dst.tile(
+                        x,
+                        y,
+                        z,
+                        tilesize=scale * 256,
+                        **layer_params.as_dict(),
+                        **dataset_params.as_dict(),
+                    )
+
+            if post_process:
+                image = post_process(image)
+
+            if rescale:
+                image.rescale(rescale)
+
+            if color_formula:
+                image.apply_color_formula(color_formula)
+
+            content, media_type = render_image(
+                image,
+                output_format=format,
+                colormap=colormap,
+                **render_params.as_dict(),
+            )
+
+            return Response(content, media_type=media_type)
+
+    # custom /statistics endpoints (remove /statistics - GET)
```

Comment:
☝️ IMO having a full dataset /statistics in a bit dangerous (as for the /preview endpoints)